lib/graph/helpers/ui5Framework.js

```diff
@@ -192,6 +192,22 @@ const utils = {
 		});
 		await projectGraph.resolveOptionalDependencies();
 	},
+	checkForDuplicateFrameworkProjects(projectGraph, frameworkGraph) {
+		// Check for duplicate framework libraries
+		const projectGraphProjectNames = projectGraph.getProjectNames();
+		const duplicateFrameworkProjectNames = frameworkGraph.getProjectNames()
+			.filter((name) => projectGraphProjectNames.includes(name));
+
+		if (duplicateFrameworkProjectNames.length) {
+			throw new Error(
+				`Duplicate framework library definition(s) found in project ${projectGraph.getRoot().getName()}: ` +
+				`${duplicateFrameworkProjectNames.join(", ")}. ` +
+				`Framework libraries should only be referenced via ui5.yaml configuration, ` +
+				`not in its dependencies (e.g. package.json). ` +
+				`Note that this error could also come from transitive dependencies.`
```

Comment:
```suggestion
				`Framework libraries should only be referenced via ui5.yaml configuration. ` +
				`Neither the root project, nor any of its dependencies should include them as direct ` +
				`dependencies (e.g. via package.json).`
```

What do you think? I'm not sure how common the term "transitive dependencies" is.